src/HDF5.jl

```diff
@@ -733,6 +733,33 @@ function h5read(filename, name::String, indices::Tuple{Vararg{Union{AbstractRang
     dat
 end
 
+"""
+    function h5read_compound(dset::HDF5.HDF5Dataset, T::DataType)
+
+Reads a dataset directly using the given datatype and returns a `Vector{T}`.
+The implementation is based on @damiendr's post in HDF5.jl Issue#408.
+"""
+function h5read_compound(dset::HDF5Dataset, T::DataType)
+    filetype = datatype(dset)  # packed layout on disk
+    memtype_id = h5t_get_native_type(filetype.id)  # padded layout in memory
+    @assert sizeof(T) == h5t_get_size(memtype_id) "Type sizes mismatch!"
+    out = Vector{T}(undef, length(dset))
```

Comment:
why not `out = Array{T}(undef, size(dset))` to preserve all the size information?

Reply:
Sure!